<HTML>
<HEAD>
<TITLE>Optimal annealing of two RNA sequences</TITLE>
</HEAD>
<BODY>
<H1> Optimal annealing of two RNA sequences </H1>
<hr>

<B>Overview:</B>
In this page you can find one of optimal annealings of two nucleic acid
sequences based on free energy calculation. <p>

<B>Instruction:</B>

All you have to do is to
enter two nucleic acid sequences in the text fields below, and press
"anneal" button. In a few minutes, you will see annealing of two
sequences in the text area below with calculated free energy.
If you enter long sequences, you have to wait for a long time to get a 
result.
<p> 

<hr>
<p>
<center>
<applet code="BasePAP.class" width="450" height="300">
Error ... cannot start this applet.
</applet></p>
</center>

<hr>
<B>Methods:</B>
Annealing is determined according to alignment of two
sequences. Alignment score is determined by free energy calculation. 
Free energy is calculated by base pairings and structural energy
required to form bulge loops and interior loops(Hairpin loops are not
considered). <p>

Energy of base pairings, bulge loops, and interior loops are calculated according to the tables below.<p> 

<CENTER>
<TABLE BORDER><CAPTION><U>Free energy of base pairing</U></CAPTION>

<TR>
<TD></TD> <TD>3'AU</TD><TD>3'UA</TD><TD>3'CG</TD><TD>3'GC</TD><TD>3'UG</TD><TD>3'GU</TD></TR>
<TR>
<TD> 5'AU </TD><TD> -0.9</TD><TD> -0.9</TD><TD> -2.1</TD><TD> -1.7</TD><TD>

-0.7</TD><TD> -0.5</TD>
</TR>
<TR>
<TD> 5'UA </TD> <TD> -1.1</TD><TD> -0.9</TD><TD> -2.3</TD><TD> -1.8</TD><TD>

-0.5</TD><TD> -0.7</TD>
</TR>
<TR>
<TD> 5'CG </TD>
 <TD> -1.8</TD><TD> -1.7</TD><TD> -2.9</TD><TD> -2.0</TD><TD>

-1.5</TD><TD> -1.5</TD>
</TR>
<TR>
<TD> 5'GC </TD>
 <TD> -2.3</TD><TD> -2.1</TD><TD> -3.4</TD><TD> -2.9</TD><TD>

-1.9</TD><TD> -1.3</TD>
</TR>
<TR>
<TD> 5'UG </TD>
 <TD> -0.7</TD><TD> -0.5</TD><TD> -1.3</TD><TD> -1.5</TD><TD>

-0.5</TD><TD> </TD>
</TR>
<TR>
<TD> 5'GU </TD>
 <TD> -0.5</TD><TD> -0.7</TD><TD> -1.9</TD><TD> -1.5</TD><TD></TD><TD>-0.5</TD>

</TR>
</TABLE><p><p>

<TABLE BORDER><CAPTION><U>Free energy of bulge loops</U></CAPTION>
<TR>
<TD></TD><TD>0</TD><TD>1</TD><TD>2</TD><TD>3</TD><TD>4</TD><TD>5</TD><TD>6</TD><TD>7</TD><TD>8</TD><TD>9</TD><TR>

<TR><TD>0</TD>  
<TD></TD><TD>  3.3</TD><TD>  5.2</TD><TD>  6.0</TD><TD>  6.7</TD><TD>  7.4</TD><TD>  8.2</TD><TD>  9.1</TD><TD> 10.0</TD><TD> 10.5</TD><TD>

</TR>
<TR><TD>10</TD>
<TD>11.0</TD><TD> 11.0</TD><TD> 11.8</TD><TD> 11.8</TD><TD> 12.5</TD><TD> 12.5</TD><TD> 13.0</TD><TD> 13.0</TD><TD> 13.6</TD><TD> 13.6</TD><TD> 

</TR>
<TR><TD>20</TD>
<TD>14.0</TD><TD> 14.0</TD><TD> 14.0</TD><TD> 14.0</TD><TD> 14.0</TD><TD> 15.0</TD><TD> 15.0</TD><TD> 15.0</TD><TD> 15.0</TD><TD> 15.0</TD><TD> 

</TR>
<TR><TD>30</TD>
<TD>15.8</TD>
</TR>
</TABLE><p><p>

<TABLE BORDER><CAPTION><U>Free energy of interior loops</U></CAPTION>
<TR>
<TD></TD><TD>0    </TD><TD>1    </TD><TD>2    </TD><TD>3    </TD><TD>4    </TD><TD>5    </TD><TD>6    </TD><TD>7    </TD><TD>8    </TD><TD>9</TD>

</TR>
<TR><TD>0</TD>
<TD>0.0</TD><TD>  0.8</TD><TD>  0.8</TD><TD>  1.3</TD><TD>  1.7</TD><TD>  2.1</TD><TD>  2.5</TD><TD>  2.6</TD><TD>  2.8</TD><TD>  3.1</TD>

</TR>
<TR><TD>10</TD>
<TD>  3.6</TD><TD>  3.6</TD><TD>  4.4</TD><TD>  4.4</TD><TD>  5.1</TD><TD>  5.1</TD><TD>  5.6</TD><TD>  5.6</TD><TD>  6.2</TD><TD>  6.2</TD>

</TR>
<TR><TD>20</TD>
<TD>  6.6</TD><TD>  6.6</TD><TD>  6.6</TD><TD>  6.6</TD><TD>  6.6</TD><TD>  7.6</TD><TD>  7.6</TD><TD>  7.6</TD><TD>  7.6</TD><TD>  7.6</TD>

</TR>
<TR>
<TD>30</TD>
<TD>  8.4<TD>
</TR>

</TABLE>

</CENTER>

Based on this free energy calculation, an optimal alignment is
obtained recursively. Dynamic programming technique is also used to
reduce an amount of calculation. <p>

<B>Applications:</B> Some biological reactions occur by base pairing of 
two sequences. For example, translation initiation site is recognized
by base pairing of 16S rRNA and Shine-Dalgarno sequence in bacteria, 
You can use this tool to characterize base pairings.<p>

Below is plots of free energy between 16S rRNA and sequences around
start codons in bacteria.
You can see apparent drops of free energies around start codons
in <I>E.coli</I>. <p>

<center>
<img src = "freeen2.gif"><p>
</center>

<B>Bibliography</B>:
Osada Y, Saito R, Tomita M(1999)
Analysis of base-pairing potentials between 16S rRNA and 5' UTR for translation initiation in various prokaryotes.
<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&cmd=Retrieve&dopt=AbstractPlus&list_uids=10487865&query_hl=2&itool=pubmed_docsum">
Bioinformatics. Jul-Aug;15(7-8):578-81
</a>
</BODY>
</HTML>

