<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type" content="text/html; charset=ISO-8859-1">
  <title>5th Annual Cytoscape Retreat and Symposium - Speaker abstracts and resumes</title>
  <link rel="stylesheet" type="text/css" media="screen" href="css/cytoretreat.css">
  <link rel="../shortcut icon" href="../images/cyto.ico">
  <!--[if lt IE 7]>
  <script defer type="text/javascript" src="js/pngfix.js"></script>
  <![endif]-->
  <script src="js/onloadheaderonly.js" type="text/javascript"></script>  
</head>

<body style="background-color: rgb(255, 255, 255);">

<!-- The header with changing pictures -->

<? include "header.php"; ?>

<!-- Retreat content table-->

<table>
 <tr valign="top">
  <!-- Left retreat navigation bar -->
  <td width="10%">
   <? include "nav_retreat.php"; ?>
  </td>
  <!-- This page content -->
  <td width="70%">
  <div id="content">
  <div class="item">
  <div class="top">
   <h2><a name="top">Symposium speaker abstracts and resumes</a></h2>
  </div>
<br>  
<p>
Find the abstracts and resumes of the speakers at the Symposium here: 
</p>
<hr>
<div class="top">
 <h2>Overview abstracts</h2>
</div>
<br>
<table class="programme">
 <tr>
  <td class="subheader" width="20%"><a href="#hood"><strong>Leroy Hood</strong></a></td>
  <td class="subheader" width="80%">Biological networks and disease</td>
 </tr>
 <tr>
  <td class="subheader" width="20%"><a href="#hopkins"><strong>Andrew Hopkins</strong></a></td>
  <td class="subheader" width="80%">Network Pharmacology: chemical opportunities for systems biology</td>
 </tr>
 <tr>
  <td class="subheader" width="20%"><a href="#versteeg"><strong>Rogier Versteeg</strong></a></td>
  <td class="subheader" width="80%">Oncogenic networks of cancer pathways in childhood cancer</td>
 </tr>
 <tr>
  <td class="subheader" width="20%"><a href="#sander"><strong>Chris Sander</strong></a></td>
  <td class="subheader" width="80%">Systems Biology of Cancer Pathways: from Molecular Perturbations to Cellular Phenotypes</td>
 </tr>
 <tr>
  <td class="subheader" width="20%"><a href="#sorger"><strong>Peter Sorger</strong></a></td>
  <td class="subheader" width="80%">Modeling Mammalian Death and Survival Pathways</td>
 </tr>
 <tr>
  <td class="subheader" width="20%"><a href="#aebersold"><strong>Ruedi Aebersold</strong></a></td>
  <td class="subheader" width="80%">Protein-centered networks in systems biology: Analysis and visualization</td>
 </tr>
 <tr>
  <td class="subheader" width="20%"><a href="#benno"><strong>Benno Schwikowski</strong></a></td>
  <td class="subheader" width="80%">Computational tools increasing sensitivity and reliability of mass spec-based proteomics</td>
 </tr>
 <tr>
  <td class="subheader" width="20%"><a href="#fekete"><strong>Jean-Daniel Fekete</strong></a></td>
  <td class="subheader" width="80%">Visualizing Dense Networks with Enhanced and Hybrid Matrices</td>
 </tr>
 <tr>
  <td class="subheader" width="20%"><a href="#birney"><strong>Ewan Birney </strong></a></td>
  <td class="subheader" width="80%">Reactome, Networks and Genomes</td>
 </tr>
 <tr>
  <td class="subheader" width="20%"><a href="#trey"><strong>Trey Ideker</strong></a></td>
  <td class="subheader" width="80%">Gaining power in gene association studies with Cytoscape</td>
 </tr>
</table>  

<hr>
<div class="top">
 <h2>Detailed abstracts and resumes</h2>
</div>

<br>
<h3><a name="hood">Leroy Hood</a></h3>
<br>

<table class="programme">
 <tr>
  <td class="subheader"><p>Position/Title/Institute</p></td>
 </tr>
 <tr>
  <td><p>President, Institute for Systems Biology</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Title</p></td>
 </tr>
 <tr>
  <td><p>Biological networks and disease</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Abstract</p></td>
 </tr>
 <tr>
  <td><p>
  I will discuss a systems approach to disease using as an example prion infection in mice.  I will show how the dynamics of protein networks derived from the dynamic analyses of brain transcriptomes superimposed on protein/protein networks from the literature and integrated together with various types of phenotypic data actually explain much of the known dynamics of the disease.  I will also show how this disease-perturbed network view of disease leads to a new approach to blood diagnostics that promises to transform early disease diagnosis and following responses to therapies.  I will also show how this approach is driving the development of new measurement technologies employing microfluidics and nanotechnology.  These systems approaches together with the new measurement technologies are driving the emergence of a new medicine that will be predictive, personalized, preventive and participatory (P4).  I will also discuss the long-term implications of P4 medicine for world health.
  </p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Biography / Awards</p></td>
 </tr>
 <tr>
  <td><p>
Dr. Hood's research has focused on the study of molecular immunology, biotechnology, and genomics. His professional career began at Caltech where he and his colleagues pioneered four instruments - the DNA gene sequencer and synthesizer, and the protein synthesizer and sequencer - which comprise the technological foundation for contemporary molecular biology. In particular, the DNA sequencer has revolutionized genomics by allowing the rapid automated sequencing of DNA, which played a crucial role in contributing to the successful mapping of the human genome during the 1990s. In 1992, Dr. Hood moved to the University of Washington as founder and Chairman of the cross-disciplinary Department of Molecular Biotechnology. In 2000, he co-founded the Institute for Systems Biology in Seattle, Washington to pioneer systems approaches to biology and medicine. Most recently, Dr. Hood's lifelong contributions to biotechnology have earned him the prestigious 2004 Biotechnology Heritage Award, and for his pioneering efforts in molecular diagnostics the 2003 Association for Molecular Pathology (AMP) Award for Excellence in Molecular Diagnostics. In 2006 he received the Heinz Award in Technology, the Economy and Employment for his extraordinary breakthroughs in biomedical science at the genetic level. In 2007 he was elected to the Inventors Hall of Fame (for the automated DNA sequencer). He has published more than 600 peer-reviewed papers, received 14 patents, and has co-authored textbooks in biochemistry, immunology, molecular biology, and genetics and is just finishing a text book on systems biology. In addition, he coauthored with Dan Keveles a popular book on the human genome project-The Code of Codes. Dr. Hood is a member of the National Academy of Sciences, the American Philosophical Society, the American Academy of Arts and Sciences, the Institute of Medicine and the National Academy of Engineering. Indeed, Dr. Hood is one of 7 (of more than 6000) scientists elected to all three academies (NAS, NAE and IOM). Dr. Hood has also played a role in founding more than 14 biotechnology companies, including Amgen, Applied Biosystems, Systemix, Darwin and Rosetta. He is currently pioneering systems medicine and the systems approach to disease. 
Awards:
<br>2005 Heinz Award for Technology,for the Economy and Employment Development and commercialization of high throughput biology (automated protein and DNA sequencing). 
<br>2003 Lemelson-MIT Prize for Invention and Innovation of an Automated DNA sequencer that made the Human Genome Project possible. 
<br>2002 Kyoto Prize for the Development of advanced biological instrumentation.
  </p></td>
 </tr>
 </table>
 
<br>
<h3><a name="hopkins">Andrew Hopkins</a></h3>
<br>

<table class="programme">
 <tr>
  <td class="subheader"><p>Position/Title/Institute</p></td>
 </tr>
 <tr>
  <td><p>SULSA Professor of Translational Biology and Chair of Chemical Informatics, University of Dundee</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Title</p></td>
 </tr>
 <tr>
  <td><p>Network Pharmacology: chemical opportunities for systems biology</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Abstract</p></td>
 </tr>
 <tr>
  <td><p>
  In recent years, it has been appreciated that many effective drugs, in therapeutic areas as diverse as oncology, psychiatry and anti-infectives, act on multiple-gene products rather than single targets. Furthermore, the advent of chemogenomics and wide ligand profiling has provided evidence that many drugs act on multiple targets: what is known as polypharmacology. This is in contrast to the predominant paradigm in drug discovery for the past two decades in which the concept of designing exquisitely selective ligands, to avoid unwanted side effects. Polypharmacology has traditionally been viewed by drug designers as an undesirable property that needs removed or reduced to produce 'clean' drugs act on single targets. However, a growing body of post-genomic biology is revealing is a far more complex picture of drug action. The combination of gene-deletion observations of phenotypic robustness and network biology theory indicate that in several instances exquisitely selective compounds may exhibit a lower than desired efficacy for the treatment of disease. Thus compounds which selectively act on two or more targets of interest could increase the confidence-in-rationale or the range of efficacy. Traditionally medicinal chemists have approached the design of ligands with multiple activities with trepidation, fearing complex structure-activity relationships or conjugated ligands with high molecular weights. Here we discuss how combining chemogenomics with network biology may enable a new 'network pharmacology' approach to drug discovery to help rationally identify compounds that act on the level of the biological network rather than a single targets, with the hope of developing more effective medicines for complex disease.
  </p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Biography / Awards</p></td>
 </tr>
 <tr>
  <td><p>
Andrew L. Hopkins is the SULSA Research Professor of Translational Biology and Chair of Chemical Informatics in the College of Life Sciences at the University of Dundee. Before taking up his appointment at Dundee, Prof. Hopkins was spent nine years in the pharmaceutical industry, his most recent position as Associate Research Fellow and Head of Chemical Genomics at the Sandwich site of Pfizer Global Research and Development. Prof. Hopkins won a British Steel scholarship to attend the University of Manchester from where he graduated with first class honours in 1993 with a First Class B.Sc.(Hons) in Chemistry. Following a brief spell in the steel industry he won a Wellcome studentship to attend the University of Oxford, working with Professor David I. Stuart FRS. He received his D.Phil. in Structural Biology from the University of Oxford in 1998. During his doctorate research Prof. Hopkins designed a new class of anti-HIV agents which where developed to drug candidates by Glaxo-Wellcome.  Following his interest in drug discovery he then joined Pfizer directly after graduating from Oxford in 1998. Over the years established various new functions for the company, including, Target Analysis in 1999, Indications Discovery in 2001 and Knowledge Discovery in 2004 and won several company awards for his efforts. Prof. Hopkins's research involves integrating chemical and biological knowledge to identify new targets or other new opportunities for medicines.  His work has involved the design and construction of informatics systems, including a hypothesis-generation system based on text-mining and ontologies and a large-scale chemogenomics integrated knowledge-base. As leader of the Indications Discovery group in he championed the initiation of several new clinical proof-of-concept studies for maraviroc, from ideas that were derived from data-mining. Prof. Hopkins is the author of over 30 scientific publications and holds 7 patents covering a diverse range of inventions, including compound design, protein engineering, new indications and informatics systems. Five of Prof Hopkins's papers have over 100 citations and 2 of which have been classified as 'hot papers' by the Thomas ISI Science Citation Index. Currently Prof. Hopkins consults for the Organization of Economic Co-operation and Development and the World Health Organization Special Programme for Training and Research in Tropical Diseases and is a member of the International Federation of Pharmaceutical Manufacturers & Associations (IFPMA) Taskforce on Partnerships. In 2007 Professor Hopkins was made a Fellow of the Royal Society of Chemistry and won the Corwin Hansch Award.  
  </p></td>
 </tr>
 </table>
 
<br>
<h3><a name="versteeg">Rogier Versteeg</a></h3>
<br>

<table class="programme">
 <tr>
  <td class="subheader"><p>Position/Title/Institute</p></td>
 </tr>
 <tr>
  <td><p>Professor in Genetics, Head Dept. of Human Genetics, Academic Medical Center, University of Amsterdam</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Title</p></td>
 </tr>
 <tr>
  <td><p>Oncogenic networks of cancer pathways in childhood cancer</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Abstract</p></td>
 </tr>
 <tr>
  <td><p>
  We study a series of human childhood tumors, with an emphasis on neuroblastomas. Only few gene defects have been identified in this embryonal tumor, but a pattern of large chromosomal deletions and gains is evident. As chromosomal copy number gains strongly disturb normal development, it is likely that the copy number defects in neuroblastoma also have strong regulatory effects. The aberrant expression of the thousands of genes in the affected regions may disturb important regulatory pathways. Identification of key-regulators in neuroblastoma would therefore require an overview of the regulatory circuits affected by chromosomal defects. With the ultimate goal of identifying drugable key-players in neuroblastoma, we therefore focus on the reconstruction of the major pathways driving this tumor. Gene expression profiles of 120 Neuroblastoma tumors and a large number of manipulated cell lines were established by microarrays. The data were analyzed in our newly developed bioinformatic tool R2. R2 detects correlations, not only between genes, gene families, pathways and chromosomal domains, but also between genes and any clinical parameter, like e.g. survival, pathology and age. The role of important regulatory genes was analyzed by identifying their downstream pathways. These genes were manipulated by inducible transgene expression, inducible siRNA knockdown or lentiviral mediated siRNA knockdown. After onset of expression or silencing, time-course experiments were analyzed by microarrays. We currently have a data set of over 800 microarrays of about 20 manipulated genes in only a few cell lines. Each of the manipulated genes typically triggered significant changes in gene expression for about 200-1000 genes. Manipulations resulting in a strong phenotype showed strong transitions in expression levels, reflecting new equilibria in the cell. Manipulation of genes from very different pathways often changed the expression of overlapping sets of downstream genes, showing a high connectivity of the regulatory networks. Integration of these connectivity loops is pursued at a second level of bioinformatics analysis, using Cytoscape as a visualization and analysis tool. Data are directly imported from R2 into Cytoscape, using in house developed plugins. In Cytoscape, pathways are reconstructed based on the time-course data and transcription factor interactions.
  </p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Biography / Awards</p></td>
 </tr>
 <tr>
  <td><p>
Rogier Versteeg did his PhD study at the Dept. of Clinical Oncology, University Medical Center Leiden (1990).From 1990 onwards, he was group-leader at the Dept. of Human Genetics et the Academic Medical Center of the University of Amsterdam. From 2001-2002 he was visiting scientist at the Max Planck Institute for Molecular Genetics, Berlin. His current position is head of the Dept. of Human Genetics (2003) and Professor in Genetics (2004) at the AMC in Amsterdam. The focus of his research is childhood cancer and genome-wide expression studies related to the higher order structure of the genome. The major focus in childhood tumors is the molecular genetics of neuroblastoma, medulloblastoma and rhabdomyosarcoma. The research on childhood tumors ranges from fundamental pathway analyses to validation of candidate drug targets. Rogier Versteeg is co-founder of the European initiative 'Innovative Therapies for Children with Cancer' (ITCC), member of the ITCC Biology Committee and in charge of the ITCC microarray profiling and biology database.  
  </p></td>
 </tr>
 </table>
 
<br>
<h3><a name="sander">Chris Sander</a></h3>
<br>

<table class="programme">
 <tr>
  <td class="subheader"><p>Position/Title/Institute</p></td>
 </tr>
 <tr>
  <td><p>Computational Biology Center - Memorial Sloan Kettering Cancer Center, New York</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Title</p></td>
 </tr>
 <tr>
  <td><p>Systems Biology of Cancer Pathways: from Molecular Perturbations to Cellular Phenotypes</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Abstract</p></td>
 </tr>
 <tr>
  <td><p>
  n.a.
  </p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Biography / Awards</p></td>
 </tr>
 <tr>
  <td><p>
Chris Sander is Head of the Computational Biology Center at Memorial Sloan Kettering Cancer Center in New York and tri-institutional professor at Rockefeller and Cornell Universities. His principal research interests are in computational and systems biology, including predictive simulations of biological processes, integrated molecular profiling of disease states, gene regulation by small RNAs, structural genomics and the development of multiplex cancer therapy. He is a leader in community efforts to create Pathway Commons, an open-source information resource for biological pathways, based on the BioPAX pathway ontology.  
  </p></td>
 </tr>
 </table>
 
<br>
<h3><a name="sorger">Peter Sorger</a></h3>
<br>

<table class="programme">
 <tr>
  <td class="subheader"><p>Position/Title/Institute</p></td>
 </tr>
 <tr>
  <td><p>Professor of Systems Biology, Center for Cell Decision Processes, Dept. of Biological Engineering, MIT and Department of Systems Biology, Harvard Medical School
</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Title</p></td>
 </tr>
 <tr>
  <td><p>Modeling Mammalian Death and Survival Pathways</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Abstract</p></td>
 </tr>
 <tr>
  <td><p>
Caspases, the proteases that dismantle apoptotic cells, normally switch from off to on in an all-or-none process that enforces an unambiguous choice between life and death.  To understand the operation of this switch in quantitative terms we have constructed a mass-action mathematical model of receptor-mediated cell death triggered by TNF and TRAIL based on known reaction pathways and trained the model on data from single cells perturbed by protein depletion, over-expression, or inhibition.  We find that receptor-mediated cell death is characterized by sudden and efficient cleavage of caspase substrates (over a 10-15 min period), but only after a remarkably long delay (1 to 12 hr), whose duration and variance depend on ligand dose and identity.  Thus, caspase regulatory pathways simultaneously achieve snap-action activation, long and variable delay and high efficiency; it is not sufficient that all processes be fast.  We hypothesize that variable delay generates a tunable dose-dependent behavior at a population level from a binary decision at a single-cell level.
<br>With the help of three newly designed single cell fluorescence reporters for initiator and executioner caspases and for mitochondrial membrane permeabilization, we have begun to dissect the trigger comprising pro-death and pro-survival BH3 proteins in the mitochondrial membrane. As suggested nearly a decade ago by Korsmeyer and colleagues, this trigger involves a competition between pore forming and pore-inhibiting proteins. While it would seem logical that the trigger would fire when activators exceed inhibitors modeling and measurement support the idea that the actual threshold is crossed at a ratio of ~0.6 to 0.9, reflecting the dynamics of the "race" between the two classes of regulators.  Direct competition between inhibitory and activating protein-protein binding makes for a very poor switch and other processes are involved in snap-action control of caspases, including slow transport of active Bax from the cytosol to mitochondria, the concentrating effects of moving from 3D to 2D diffusion and high-order assembly of Bax-containing pores.
<br>Once mitochondria depolarize a powerful feed-forward circuit, involving cytosolic translocation of SMAC/DIABLO and cytochrome C leads to activation of executioner caspases. Prior to that point, for up to 12 hr in the cases of low TRAIL doses active initiator caspases accumulate steadily, leading to cleavage of Caspase 3 when is then strongly repressed. The generation of such a potent and lethal protease in cells that not yet committed to death is remarkable.  Moreover, under some conditions, it gives rise to unexpected failure modes in which cells commit to partial death and survive with damage to their genomes.  Such an event is expected to cause genomic instability, consistent with previous suggestions that the chromosomal translocations typical of lymphoma might be caused by failures of apoptosis.
  </p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Biography / Awards</p></td>
 </tr>
 <tr>
  <td><p>
Peter Sorger, Ph.D is a Professor of Systems Biology at Harvard Medical School and holds a joint appointment in MIT's Dept. of Biological Engineering and Center for Cancer Research. He received his A.B. from Harvard College, Ph.D. from Trinity College Cambridge, U.K. and trained as a postdoctoral fellow with Harold Varmus and Andrew Murray at the University of California, San Francisco. Sorger was co-founder of the MIT systems biology program CSBi, Merrimack Pharmaceuticals and Glencoe Software and serves on the scientific advisory and corporate boards of several other technology companies. He is currently Chair of the CSF study section of the NIH and Director of the NIH-NIGMS CDP Center for Systems Biology.  
  </p></td>
 </tr>
 </table>
 
<br>
<h3><a name="aebersold">Ruedi Aebersold</a></h3>
<br>

<table class="programme">
 <tr>
  <td class="subheader"><p>Position/Title/Institute</p></td>
 </tr>
 <tr>
  <td><p>Institute of Molecular Systems Biology, ETH - Swiss Federal Institute of Technology Zurich
</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Title</p></td>
 </tr>
 <tr>
  <td><p>Protein-centered networks in systems biology: Analysis and visualization</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Abstract</p></td>
 </tr>
 <tr>
  <td><p>
   n.a.	  
  </p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Biography / Awards</p></td>
 </tr>
 <tr>
  <td><p>
Prof. Ruedi Aebersold is one of the pioneers in the field of proteomics. He is known for developing a series of methods that have found wide application in analytical protein chemistry and proteomics like a new class of reagents termed Isotope Coded Affinity Tag (ICAT) reagents used in quantitative mass spectrometry. Prof. Dr. Aebersold and his team of researchers use the protein profiles determined by this method to differentiate cells in different states, such as non-cancerous versus cancerous cells, and to systematically study how cells respond to external stimuli. These "snapshot" profiles indicate which cells contain abnormal levels of certain proteins. This is expected to lead to new diagnostic markers for disease and to a more complete understanding of the biochemical processes that control and constitute cell physiology.
<br>Prof. Aebersold serves on the Scientific Advisory Committees of numerous academic and private sector research organizations and is a member of several editorial boards in the fields of protein science, genomics, and proteomics.
<br>Prof. Aebersold is a native of Switzerland and obtained his Ph.D. in Cellular Biology at the Biocenter of the University of Basel in 1983. Since that time, he is a faculty member of the Universities of Washington and British Columbia, until 2000, when he co-founded the Institute for Systems Biology in Seattle. In 2004, he accepted a position as full professor at the Institute of Biotechnology at the Swiss Federal Institute of Technology (ETH) in Zurich, where in January 2005, his research group became the first integral part of the newly founded Institute of Molecular Systems Biology.
  </p></td>
 </tr>
 </table>
 
<br>
<h3><a name="benno">Benno Schwikowski</a></h3>
<br>

<table class="programme">
 <tr>
  <td class="subheader"><p>Position/Title/Institute</p></td>
 </tr>
 <tr>
  <td><p>Chef du Laboratoire, Systems Biology Group, Institut Pasteur, Paris
</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Title</p></td>
 </tr>
 <tr>
  <td><p>Computational tools increasing sensitivity and reliability of mass spec-based proteomics</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Abstract</p></td>
 </tr>
 <tr>
  <td><p>
   Mass spectrometry is one of the main technologies to survey molecular systems on the protein level. Despite the high sensitivity of mass spectrometry, the practical sensitivity and reproducibility of this technology is still limited in typical mass spec workflows. In this presentation, I will explain the nature of these limitations, and current developments to bypass them - specifically, the computational alignment of multiple experiments, and the automated visualizations to analyze the reproducibility of the analytical platform.	  
  </p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Biography / Awards</p></td>
 </tr>
 <tr>
  <td><p>
Trained as a mathematician and computer scientist (Ph.D. Univ. Bonn, Germany), and after holding a position as assistant professor at the Institute for Systems Biology in Seattle, Dr. Schwikowski is now leading the Systems Biology Group at Institut Pasteur in Paris, France. Dr. Schwikowski has made original contributions to a number of areas in classical bioinformatics, such as sequence analysis and phylogenetics, and DNA array design. After demonstrating, in 2000, the significance of large-scale protein interaction data for functional genomics and systems biology, he co-founded, in 2001, the Cytoscape project (with Trey Ideker). His group is now dedicated to the development of integrative approaches to biology by developing mass spectrometry algorithms for proteomic analyses, and data integration approaches that turn current large-scale data into biological insight.
  </p></td>
 </tr>
 </table>
 
<br>
<h3><a name="benno">Jean-Daniel Fekete</a></h3>
<br>

<table class="programme">
 <tr>
  <td class="subheader"><p>Position/Title/Institute</p></td>
 </tr>
 <tr>
  <td><p>Research Director - Head of the AVIZ Team - INRIA, France
</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Title</p></td>
 </tr>
 <tr>
  <td><p>Visualizing Dense Networks with Enhanced and Hybrid Matrices</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Abstract</p></td>
 </tr>
 <tr>
  <td><p>
   The Node-Link Diagram is the most popular visual representation for networks. However, when a network becomes dense or large, they become unreadable. The Matrix representation is known but not used by most network visualization systems. We showed in 2003 that it outperforms the Node-Link representation for large and dense networks for most low-level tasks, except path-related ones.
<br>In this talk, we present the state of our research regarding the enhancement of visual matrices including MatrixExplorer, a dual-representation system for exploring large networks, MatLink, an augmented matrix representation and NodeTrix, a hybrid representation combining node-links and matrices.
<br>We also argue for the addition of these representations in network visualization systems to complement and augment their analytical and exploration power.	  
  </p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Biography / Awards</p></td>
 </tr>
 <tr>
  <td><p>
Jean-Daniel Fekete received his PhD from Univ of Paris 11 in 1996. He headed the "Interactive Design and Modeling" group of the Ecole des Mines in Nantes in 2000, was Invited Researcher at the University of Maryland at College Park in 2001-2002. He is a researcher at INRIA since 2002, where he heads the AVIZ Research team since 2007. His research interests include Human-Computer Interaction, Information Visualization and Visual Analytics. He authored 70 articles in journals and conferences. He is editor of the "International Journal of Human-Computer Studies" and co-founder of the Information Visualization Contest, an event that takes place every year during the InfoVis Conferences since 2003.
  </p></td>
 </tr>
 </table>
 
<br>
<h3><a name="birney">Ewan Birney </a></h3>
<br>

<table class="programme">
 <tr>
  <td class="subheader"><p>Position/Title/Institute</p></td>
 </tr>
 <tr>
  <td><p>Senior Scientist, European Bioinformatics Institute
</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Title</p></td>
 </tr>
 <tr>
  <td><p>Reactome, Networks and Genomes</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Abstract</p></td>
 </tr>
 <tr>
  <td><p>
Modern genomics provides a variety of datasets to understand how our genome functions, from chip-chip, chip-seq, protein-interactions and protein pathways. I will illustrate how at the EBI we have integrated these datasets to provide more insights into how the human genome functions.
  </p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Biography / Awards</p></td>
 </tr>
 <tr>
  <td><p>
Dr Birney is a Senior Scientist at EMBL working at the EBI. Best known as the head of the EBI side of the Ensembl project, his group has recently merged with Rolf Apweiler's group to form the large "PANDA" group (Protein and Nucleotide Databases). Dr Birney originally trained as a Biochemist, but moved quickly into Bioinformatics. His first set of programs (Pairwise and Searchwise) were published while an undergraduate at Oxford during which time he worked with Adrian Krainer's Lab (at CSHL), Toby Gibson (at EMBL) and Iain Campbell's lab at Oxford. His PhD was with Richard Durbin at the Sanger Institute, and ha has collaborated with him since. In 2000 Dr Birney joined the EBI as a Team Leader, and I is now a Senior Scientist in EMBL (EMBL is the parent organisation of EBI) and part of the EBI Senior Management. Dr Birney has worked with many genome projects (in particular Human, Mouse and Chicken), has been an active member in the Bioperl project and has previously worked with the Pfam project.
  </p></td>
 </tr>
 </table>
 
<br>
<h3><a name="trey">Trey Ideker</a></h3>
<br>

<table class="programme">
 <tr>
  <td class="subheader"><p>Position/Title/Institute</p></td>
 </tr>
 <tr>
  <td><p>Associate Professor of Bioengineering, UCSD
<br>President of Cytoscape Consortium
</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Title</p></td>
 </tr>
 <tr>
  <td><p>Gaining power in gene association studies with Cytoscape</p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Talk Abstract</p></td>
 </tr>
 <tr>
  <td><p>
Gene association studies seek to identify significant links between a pattern of genetic markers (e.g., a panel of SNPs constituting an individual's haplotype) and the incidence of disease (e.g., as measured using the technique of Quantitative Trait Loci or QTLs).  Due to the spacing of markers or the effects of linkage disequilibrium, each marker may be near many genes making it difficult to finely map the causal factors responsible for the observed disease phenotype.  To address this challenge, we have developed a network-based analysis platform in Cytoscape called 'eQTL Electrical Diagrams' (eQED).  This approach integrates QTLs and eQTLs with protein interaction networks by modeling the two data sets as a wiring diagram of current sources and resistors.  When used to analyze a large eQTL study in yeast, we show that eQED correctly identifies 79% of known regulator-target pairs in yeast, which is significantly higher performance than three competing methods.  eQED also annotates 368 protein-protein interactions with their directionality of information flow with an accuracy of approximately 75%.
  </p></td>
 </tr>
 <tr>
  <td class="subheader"><p>Biography / Awards</p></td>
 </tr>
 <tr>
  <td><p>
The Ideker laboratory is internationally recognized for its work in genome-scale mapping and assembly of cellular models.  In 2001, Ideker and Lee Hood first defined the field of "Systems Biology" in a highly-cited article in Science and an associated commentary published in Annual Reviews.  While on the faculty at UCSD, Ideker has pioneered Comparative Network Analysis as a major emerging area within biology and computer science, involving integration and alignment of protein network maps across species, biological stimuli, disease states, and network types.  The Ideker laboratory has recently applied these methods to construct the first system-wide model of the cellular response to DNA damage, which the lab is now using to identify new cancer biomarkers and therapeutics.  This and other recent work has appeared in journals including Science, Nature, Nature Biotechnology, Nature Molecular Systems Biology, Genome Research, Bioinformatics, and PNAS, and the lab's research has also been featured in popular news outlets such as Forbes magazine, The Scientist, and the San Diego Union Tribune.  
<br>Ideker has been twice featured by Technology Review magazine, in 2005 as one of the 35 top scientists under 35 years old, and in 2006 as having developed one of the ten most promising new technologies overall.  Ideker serves as Associate Editor of the journal Bioinformatics and chairs the successful RECOMB Systems Biology conference annually. Ideker is Chairman of the Cytoscape Consortium, an international collaboration of eight universities and research institutes that develop and promote Cytoscape, an open-source platform for protein network modeling and visualization which is the industry standard.
  </p></td>
 </tr>
 </table>

  </div>
  </div>

  </td>  <!--End Main page content -->

  <!-- Sponsors box -->
  <td width="20%">
   <? include "hosted_box.php"; ?>
   <? include "sponsors_box.php"; ?>
  </td>
 </tr>
</table>
<? include "footer.php"; ?>
</body>
</html>
