In this section, mechanisms of translation initiation discovered by 
other researches are described.
The process of translation initiation in eucaryotes is different from that
of procaryotes.
We first discuss the translation initiation process of procaryotes and
then eucaryotes.


\subsection{Translation initiation in procaryotes}

In procaryotes, a ribosome with tRNA which carries methionine binds to 
the specific region of mRNA and recognizes AUG codon nearby
and protein synthesis begins. In this process the main factors are
ribosome, tRNA with methionine(fMet-tRNA$^{\rm Metf}$), and mRNA. In addition, 
at least initiation factors and GTP molecule are required to ensure
the efficiency and fidelity of this process.

The descriptions of each factors are given\cite{label20}.


\subsubsection{fMet-tRNA$^{\rm Metf}$}
A translation starts with methionine, whose codon is usually AUG.
Initiator tRNA(tRNA$^{\rm Metf}$) with methionine(fMet-tRNA$^{\rm Metf}$)
recognizes AUG codon and translates it to 
methionine.
AUG codon in the middle of the coding region is translated by 
Met-tRNA$^{\rm Met}$ which
is used for elongation of protein. fMet-tRNA$^{\rm Metf}$, which is used for 
translation
initiation, is different from Met-tRNA$^{\rm Met}$ in some aspects.
One of these is the presence of three consecutive GC base pairs in the 
anticodon stem which is important for the formation of anti-codon loop
and binding to the ribosomal P-site\footnote{P-site is where the end of
elongating amino acid with tRNA(peptidyl-tRNA) is located, 
waiting for the next amino acid with tRNA(aminoacyl tRNA)}.
Another difference is that \(\alpha\)NH2 group of the methionine in
 fMet-tRNA$^{\rm Metf}$ is blocked by formylation and it cannot form a peptide
bond in here. It means that it cannot work as elongation tRNA.

\subsubsection{Ribosome}
Bacterial ribosome which is used for translation is 70S\footnote{S values 
indicate the rate of sedimentation in an ultracentrifuge}.
It consists of 30S small subunit and 50S large subunit.
Furthermore, 30S small subunit contains 16S rRNA and 21 proteins, and
50S large subunit contains 5S rRNA, 23S rRNA and 34 proteins.
Only after the small ribosomal subunit loaded with initiation factors finds
the start codon does the large subunit bind.


\subsubsection{mRNA}
Bacterial mRNAs are commonly polycistronic. That means that they encode
multiple proteins that are separatedly translated from the same mRNA
molecule. Sometimes coding regions overlap, but it may not effect the
fidelity of
translation. Sometimes coding regions overlap by one base, which 
will be like [UG\underline{A]UG}.

In bacterial mRNA, ribosome binding site and start codon play important
roles for translation initiation. Ribosome binding site is where the 30S
small subunit binds first on mRNA. This site contains purine\footnote{Adenine
and guanine} rich sequence which is called Shine-Dalgarno sequence
\cite{label7}. The 3' terminal of 16S rRNA in 30S subunit binds
to this sequence and helps 30S subunit to bind to mRNA.
This interaction makes a major contribution to the efficiency of initiation
and provides the bacterial cell with a simple way to regulate protein
synthesis. Many translational control mechanisms in procaryotes involve 
blocking the Shine-Dalgarno sequence, either by covering it with a bound
protein or by incorporating it into a base-paired region in the mRNA
molecule. 

Most {\it E.Coli} mRNAs have Shine-Dalgarno sequences(except mRNA for dnaG
primase).
The sequence of 3' terminal of 16S rRNA in {\it E.Coli} is
"....acctgcggttggatcacctcctta". Shine-Dalgarno sequence should be
complement to this sequence, i.e., "taaggaggtgatccaaccgcaggt...." .
However, this sequence is not usually conserved, and only 3 to 9
nucleotides will make pairs with the 16S rRNA 3'terminal sequence.
If the length of complementarity is too long, ribosome will bind
to mRNA too tightly, disturbing the migration of the ribosome.
There are homology between {\it E.Coli} 16S rRNA and that of other bacteria.

Schneider et al. has analyzed translation initiation sites
in terms of information
theory\cite{label11}. And Barrick et al. has quantitatively analyzed 
Shine-Dalgarno sequence by the experiments\cite{label28}.
There is a possibility that 16S rRNA binds to
multiple regions of mRNA. This may be upstream of AUG, the spacer region 
between Shine-Dalgarno sequence and AUG, or downstream of AUG\cite{label10}.
But for downstream of AUG, the position +16 has been determined by experiments
as being the limit of the downstream region in the mRNA which interacts 
strongly with the ribosome.
I.G.Ivanov et al. has found a second putative
mRNA binding site on {\it E.Coli} 16S rRNA\cite{label23}.


 Distance between the Shine-Dalgarno sequence and
the start codon is known to affect efficiency of translation
significantly. 
Optimal distance between the Shine-Dalgarno sequence and the start codon was
estimated to be 4 to 9 base pairs\cite{label9}.
Three examples are listed below. The Shine-Dalgarno sequence and functional
initiator codon are shown in larger type, and the underlined AUG and
GUG triplets are not functional, presumably because they lie too close
to the Shine-Dalgarno sequence\cite{label22}.


\begin{description}
\item[{\it E.Coli} trpC]  {\large GAGG}GUA\underline{AUG}{\large AUG}
\item[coliphage QB polymerase]
{\large UAAGG\underline{A}}\underline{UG}AA\underline{AUG}C{\large AUG} 
\item[{\it E.Coli} lac I] 
{\large G\underline{GUG}}\underline{GUG}A\underline{AU{\large G}}{\large UG}
\end{description}




Start codon is AUG in most of the time, but sometimes GUG(8\%) and UUG(1\%)
work as start codon. In most cases, changing the rare initiation triplet
into the more common AUG will cause the increase of expression and this
may disturb the normal mechanism of gene expression. Thus one of the reasons
that the bacteria use start codon like GUG or UUG is that they may be
appropriate for the expression control of some specific genes. Notice that
when GUG is read as a start codon, it is translated to formyl-methionine.
If we change normal AUG to GUG, translation may not occur.


AUG triplets preceded by appropriately
spaced Shine-Dalgarno-like sequences appear randomly throughout the
{\it E.Coli} genome, which does not function as translation initiation site.
Thus, besides Shine-Dalgarno sequence and start codon and their distance,
there are some factors that affect translations. The important one is 
a secondary structure. It may separate or hide Shine-Dalgarno sequence
and/or AUG triplet.
Thus, an access of ribosome to non-initiation site of mRNA is restricted by the
secondary structure of mRNA. But a good Shine-Dalgarno complementarity
provides the ribosome with an increased affinity for its binding site,
and thereby enhances its ability to compete against the secondary 
structure\cite{label26}.

Another factors may be some consensus sequences. That are sequences 
preceding Shine-Dalgarno sequence,
sequences between Shine-Dalgarno sequence and start codon, and sequence
following the start codon. 
\begin{enumerate}
\item Except for ribosome binding sites, there are not many  G's and there are
many A's and U's.
\item position -3 is likely to be A and position +4 through +7 is
likely to be GCUA or AAAA. In one experiment, which changed AAA to AAG,
the translational efficiency reduced by 70\%\cite{label25}.
\item When the sequences preceding Shine-Dalgarno sequence is eliminated,
sometimes translation does not occur.

\end{enumerate}



 
\subsubsection{Initiation factors}
The initiation process is complicated, involving a number of steps catalyzed
by proteins called initiation factors, many of which are themselves
composed of several polypeptide chains. In {\it E.Coli}, three initiation 
factors namely IF1, IF2 and IF3 seem to play important roles for translation 
initiation. 30S subunit has a single high-affinity site for each factor.
But IF1 and IF3 show little affinity for 50S subunit and for 70S monomers.
In fact, they are ejected from the 30S subunit when 30S and 50S subunit
associate. But IF2 has fairly high affinity for 50S subunit and 70S monomers.

The main functions of these initiation factors are to control associations
and disassociations of codons and anti-codons of tRNA in P-site of the
ribosome and, ultimately, to influence which and how many 30S initiation 
complexes enter the elongation cycle after association with the 50S subunit.

Overview of functions of each initiation factors are listed below.
\begin{description}

\item[IF1] Main functions are obscure. 
It may be involved in some stabilization.

\item[IF2] It binds to 30S subunits. It contains a binding site for fMet-tRNA.
      It effects the formation of initiation complexes(which is an association
of  ribosome, mRNA, and some other factors)kinetically. Upon subunit 
interactions, it interacts with 50S subunit, activating enzyme for GTP.
It positions fMet-tRNA in P-site.

\item[IF3] It binds to 30S subunits and is ejected upon subunit association.
  When it is binding to 30S subunit, 50S subunit cannot associate. It is
  needed when 30S subunit binds to initiation site of mRNA. But it seems that
  it is not involved in the decision of translation start site.
  IF3 crosslinks to 16S rRNA in two regions in one time.\\

\end{description}
These three initiation factors do not share any structural homology
and appear to be evolutionarily conserved. For example, IF1 of {\it E.Coli}
shares 69\% of identical residues with IF1 of {\it B.subtilis}, while IF3
of {\it E.Coli} is 50\% identical with IF3 of {\it B.stearothermophilus}.

\begin{figure}
\begin{center}
\epsfile{file=transbct.eps,scale=0.8}
\end{center}
\caption{Simplified mechanistic model of translation initiation in
procaryotes}
\end{figure}



\subsection{Translation initiation in eucaryotes}
In eucaryotes, small subunit of ribosome first binds to the 5' end of
mRNA and then moves downward to the 3' direction by the hydrolysis of 
ATP, searching
for translation initiation site instead of
directly forming complexes around start codon like bacteria. 
In this process, the main factors are ribosome, tRNA with 
methionine(Met-tRNA$^{\rm Meti}$), and mRNA. In addition, many initiation factors 
are needed. It is far more than that of procaryotes.

\subsubsection{Met-tRNA$^{\rm Meti}$}
Like bacteria, the translation start with methionine.
In case of eucaryotes, tRNA$^{\rm Meti}$ is used for translation initiation.
It binds to methionine to form Met-tRNA$^{\rm Meti}$. Unlike bacteria,
this methionine is not formylated.

\subsubsection{Ribosome}
Eucaryotic ribosome which is used for translation is 80S. It consists
of 40S small subunit and 60S large subunit. Furthermore,
40S small subunit contains 18S rRNA and about 30 proteins,
and 60S large subunit contains 28S rRNA, 5.8S rRNA, 5S rRNA and
about 40 proteins. It is 40S subunit that binds to mRNA to scan for
translation initiation site. There are homology between 18S rRNA and
16S rRNA of {\it E.Coli}.

\subsubsection{mRNA}
Eucaryotic mRNA do not have Shine-Dalgarno sequences. Instead,
it has CAP structure. The CAP structure is a structure in
which phosphoric acid is attached to 5' terminal of mRNA
and it plays an important role for translation initiation. 
It is recognized by 40S subunit and 40S subunit binds to it.
5'UTR(UnTranslated Region) of vertebrate mRNA is usually GC rich.
It helps mRNA to form secondary structure by GC base pairings.
Secondary structure may disturb translation as it will be
described later. This means that expression of gene is controlled 
at the level of translation by this feature. Those mRNAs that are 
especially GC rich in 5'UTR are likely to be undertranslated when the general 
translational capacity declines. The translation of most cellular 
mRNAs is, in fact, inhibited by serum deprivation, or heat shock, or
virus infection. Actual GC content in 5'UTR partly depends on the kind of
protein that the mRNA makes. But most mRNAs have GC content higher
than 50\%. 

Start codon is AUG most of the time, but some experiment showed
that translation can be initiated from non-AUG codons, such as
ACG, CUG, or GUG\cite{label6}\cite{label29}.


\subsubsection{Initiation factors}
One of the main initiation factors in eucaryotes is eIF and there are
many kinds of eIF. Nine initiation factors of reticulocyte and its
functions are listed in table \ref{eif}.
\begin{table}
\begin{tabular}{|c|l|}
\hline
factor & ~~~~~~~~~~~~~~~~~~function\\
\hline
eIF3 & binds to mRNA\\
CBPI,CBPII & help ribosome to bind to CAP of mRNA and break secondary structure \\
eIF1,eIF4B & help binding of mRNA\\
eIF4A & helps binding of mRNA and binds to ATP\\
eIF6 & disturbs associations of 40S and 60S ribosomal subunits\\ 
eIF5 & separates eIF2 from eIF3\\
eIF4C & binds to 60S ribosomal subunit\\
eIF2 & binds to Met-tRNA$^{\rm Meti}$\\
eIF4D& unknown \\
\hline
\end{tabular}
\caption{Initiation factors in eucaryotes}
\label{eif}
\end{table}
Hannig identified yeast GCD10 gene as the structural gene for the yeast
eIF3 and analysis of mutant phenotypes has opened the door to the genetic
dissection of the eIF3 protein complex\cite{label24}.

\begin{figure}
\begin{center}
\epsfile{file=transeuc.eps,scale=0.8}
\end{center}
\caption{Simplified initiation cycle on ribosomes in eucaryotes}
\end{figure}

\subsubsection{Decision of translation initiation site}
In the "scanning model", ribosomal 40S subunit first binds to the CAP
structure and then moves downward to the 3' direction.  It is proved
by experiments that ribosome can not go back to 5'
direction\cite{label1}. Most of the time, ribosome uses the first
found AUG as a starting AUG for translation initiation.  However, it
sometimes passes this first AUG and looks for another AUG for
translation initiation.  This process is called "leaky
scanning"(figure \ref{dscr4}).

\begin{figure}
\begin{center}
\epsfile{file=dscr4.ps,hscale=0.85,vscale=0.75}
\end{center}
\caption{Leaky Scanning}
\label{dscr4}
\end{figure}

We can think of four possible causes in order for the leaky scanning to
occur.

\begin{itemize}
\item Nucleotide sequences around the AUG are not suitable for
initiation. \\
The preferred sequence around a starting AUG in
vertebrates, for example, is gccGCCACC[AUG]G\cite{label3}.

Position -3\footnote{Traditionally 
position +1 is defined as A-residue of the AUG initiation codon. The adjacent
nucleotide 5' with respect to this A-residue is defined as position -1
 Ex.C$^{-2}$
C$^{-1}$$A^{+1}T^{+2}G^{+3}$G$^{+4}$. But in this paper,
sometimes the position where A-residue of start ATG is located is defined 0.}
is often adenine or guanine\cite{label4};AUGs are likely to be skipped
if its -3 position is cytosine or thymine. But if guanine is
located in position +4 in this case, it may have some effect to
prevent the leaky scanning. 

\item The AUG trinucleotide is followed shortly by a stop codon.\\
Earlier studies showed that
ribosomes can reinitiate the translation from the second AUG 
if the first AUG is followed shortly by a stop codon in frame. Initiation 
factors dissociate from the 40S ribosomal subunit upon the addition of 
a 60S ribosomal subunit and initiation of peptide bond formation. But
if the 40S reaches stop codon right after the translation initiation, 
initiation factors are still on the 40S and it has an ability to
reinitiate translation\cite{label4}. The
efficiency of reinitiation increases when stop codons are located far
upstream of the start codon\cite{label18}.

\item Distance between the CAP structure and the AUG is too
short.\\  It has been shown that the leaky scanning is likely to occur
when the distance between the CAP structure and the AUG trinucleotide is too
short\cite{label19}.  
Some organisms take advantage of this characteristics to produce
alternative proteins from a single gene by alternating
the location of the CAP structure (i.e., transcription initiation site),
resulting alternating selection of a start codon.
\cite{label5}.

\item Secondary structure of mRNA affects translation initiation.\\
Secondary structures such as hair pin loops in mRNA may disturb the 
recognition of starting AUG\cite{label21}. Advancing on mRNA, 
40S subunit breaks the secondary structure of mRNA if the stability
of the secondary structure is below \(\Delta\)G = -30\footnote{Free
energy}. But if it has 
more stability, advancing subunit will slow down or stop, which may
disturb the translation initiation.
But in some cases, it may enhance the recognition of the 
starting AUG\cite{label2}. For example, when the bases around AUG
in mRNA are not in a favorable context, subunit is likely to pass by. 
But if the mRNA has hairpin structure downstream of this AUG triplet,
migrating ribosome may slow down, having more time to recognize AUG
as a start codon.
\end{itemize}

Note that a few eucaryotic mRNAs initiate translation by the different
mechanism 
rather than scanning. These mRNAs contain complex nucleotide
sequences called internal ribosome entry sites. Ribosomes bind here
without the help of CAP and start translation at the next AUG codon
downstream\cite{label0}.  

\subsection{Translation initiation in mitochondria}
Mitochondria are membrane-bounded organelles(organs that have specific
functions in the cell) that carry out oxidative phosphorylation and
produce most of the ATP in eucaryotic cells. They have their own
genetic systems;They have their own DNA and produce some proteins
of their own. 

Translation often starts from the site close to the transcription 
initiation site. 
For the start codon of mitochondria, AUG is used frequently. In addition, AUA
and AUU are also used.  

