
\section{Overview}

Base correlations surrounding translation initiation sites are examined. 
In most bacteria, there were base correlations that are likely to form
Shine-Dalgarno sequences. However, clear tendency to form consensus
sequences were not observed in eucaryotes. This may suggest that in
eucaryotes in general, rRNA does not bind to consensus sequence on mRNA
to initiate translation. In {\it H.sapiens} and {\it M.musclus}, there
were correlations between -2 and -1 that are likely to make bases at
these positions the same. Further analyses have showed that there are
some correlations between Shine-Dalgarno sequences and first bases of
start codons. There were no strong correlation between upstream region
and downstream region.


\section{Introduction}

Sequence patterns surrounding start codons are very
important for ribosomes to recognize translation initiation
sites. Thus, determining consensus patterns is very important, and
there are some works that analyzed these patterns. Finding consensus
means evaluating how bases are {\bf conserved}. However, there are
factors other than conservation that are hidden in sequences. One of
that is {\bf correlations}. 

Suppose that a single base ``a'' at position P1 has influence +X on
translation initiation. Simultaneously, a single base ``t'' at
position P2 has influence +Y on translation initiation. If there is
``a'' at position P1 and ``t'' at position P2, how much influence
would they have on translation initiation? If bases at position P1 and
P2 work independently, the influence would be close to X +
Y. However, if ``a'' at position P1 and ``t'' at position P2 have
interactions, amount of influence would be very different from X + Y,
possibly much larger or much less than X + Y. In this case, we say
that there is correlation between position P1 and P2. But, how can we
assess these correlations?

One of the ways is to calculate whether observing base b1 at position
P1 has any influence on probability of observing base b2 at position
P2, provided that correlation of bases on translation initiation also
has correlation on those probabilities. Based on this measurement, it
is possible to assess the amount of correlations. The exact way is
discussed in \ref{cor_mat_meth}.

If correlation between bases are observed, it can be assumed that
bases work dependently as recognition sequences. For example,
Shine-Dalgarno sequences are recognized by 3' terminal of 16S
rRNA. Single base does not seem to be able to contribute to
interaction between 16S rRNA and Shine-Dalgarno sequence, because
single base does not have ability to contribute to those bindings. Two
or more bases may contribute to those bindings and thus probability of
observing ``g'' after ``a'' is expected to be very large since the
core Shine-Dalgarno sequence is AGGAGG. In case of Kozak's consensus
sequence, it is difficult to infer whether there is correlation or
not, because what kind of interaction that will occur at Kozak's
consensus sequence at the time of translation initiation is not clear,
and it is worth investigating.

In this work, comprehensive analyses of correlations in translation
initiation sites were conducted to investigate what kind of correlations
exist in this region.

Correlations hidden in DNA sequences are previously investigated by many
researchers. There are short range correlations, i.e., 1 or several
bases, and long range correlations, i.e, 10 or more, in the
genomes. Correlations between adjacent bases, or dinucleotide or
trinucleotide biases in other words, are investigated by Karlin et
al.\cite{label301}, for example. They have described that there are
correlations between adjacent positions throughout the genome in many
species, such as low existence of G after C in many
eucaryotes. According to their research, there seems to be more than
first order of Markov chain in DNA sequences in many organisms.

Long range correlations are investigated by Herzel et
al.\cite{label511}, for example. They showed that there is long range
correlation of G and C that decay slowly, probably because of difference
in GC content in coding and non coding region, Alu repeats, L1 repeats
or CpG islands. Also they showed that there is correlation in DNA
sequences with period of 3, which is probably the consequence of frame
in the coding region. In addition, some previous research have shown
that there are correlations with periodicity of 10-11 base pairs in some
genomes\cite{label511,label509}.

However, there are only few works that evaluated correlation hidden in
translation initiation sites statistically. Some analyses have been done
with Hidden Markov Models\cite{label515}, but those are usually
artificial.

\section{Materials and Methods}
\label{cor_mat_meth}

There are several ways to measure the correlations between adjacent
bases, or bases separated by other several bases, and they will be
discussed.

\subsection{Statistical test on first Markov process}

Recall that genome has dinucleotide or trinucleotide
biases\cite{label301}. Calculating trinucleotide or more biases
requires much more data. Thus, only dinucleotide biases are considered. 
In this regard, amount of correlation is amount of how much biases in
translation initiation sites differ from biases in the genome.

To formulate it statistically, let \(R(b1)\) be probability of observing
base b1 in the genome, \(R(b2|b1)\) be probability of observing base b2
just after b1 in the genome, \(P(b1, b2)\) be probability of observing
b1 and b2 in adjacent positions in translation initiation sites.

The null hypothesis is:

\[
 P(b1, b2)=R(b2|b1)R(b1) \quad \mbox{for all }b1, b2 = a,t,c,g
\]



And the alternative hypothesis is

\[
 P(b1, b2) \neq R(b2|b1)R(b1) \quad \mbox{for some }b1,b2 = a,t,c,g
\]

Assuming that null hypothesis holds, state
transition matrix of bases can be expressed by the following matrix
\(M\).

\[ M = \left(
 \begin{array}{cccc}
 P(a|a) &  P(t|a) &  P(c|a) &  P(g|a) \\ 
 P(a|t) &  P(t|t) &  P(c|t) &  P(g|t) \\ 
 P(a|c) &  P(t|c) &  P(c|c) &  P(g|c) \\ 
 P(a|g) &  P(t|g) &  P(c|g) &  P(g|g) \\ 
 \end{array}
 \right) 
\]

where \(P(b2|b1)\) denotes the probability of having base \(b2\) at
position n+1 provided that there is base \(b1\) at position n. This is
the case where two positions are next to each other. In case where
those two positions are separated by \(n\) bases, state transition
matrix \(M_{n}\) is

\[
 M_{n} = M^{n}
\]

However, \(P(b2|b1)\)'s are not equal all over the genome. In fact, at
least, it is different in non coding regions, and first-second bases,
second-third bases, and third-first bases in codons in coding
regions\cite{label301}. The corresponding matrix \(M_{(nc)}\),
\(M_{(12)}\), \(M_{(23)}\) and \(M_{(31)}\) must be pre-calculated. Then
matrix between position p1 and p2 is

\[ 
_{p1}M_{p2} = \prod_{n=p1}^{p2-1}M_{(n)}
\]

\noindent
where
\[M_{(n)} = \left\{
   \begin{array}{ll}
 M_{(nc)}, & \quad \mbox{position}\:n\:\mbox{being non coding region}\\
 M_{(12)}, & \quad \mbox{position}\:n\:\mbox{being first position of the codon}\\
 M_{(23)}, & \quad \mbox{position}\:n\:\mbox{being second position of the codon}\\
 M_{(31)}, & \quad \mbox{position}\:n\:\mbox{being third position of the codon}\\
   \end{array}
   \right.
\]

\vspace{1em}

To evaluate statistical significance, the following formula is used(See appendix \ref{cor_test}).

\[
2N\sum_{i=a,t,c,g}\sum_{j=a,t,c,g}
                        P(i,j)\log\frac{P(i,j)}{P(i)P(j|i)} \quad \sim \chi^{2} \:\: \mbox{with 9 degrees of freedom}
\]

\noindent
where \(N\) denotes number of sequences used for the analysis,
\(P(i)\) denotes probability of observing base \(i\) at position p1,
\(P(i,j)\) denotes probability of observing base \(i\) at position p1 
and base \(j\) at position p2 at the same time. And \(p(j|i)\) denotes
corresponding element of \(_{p1}M_{p2}\).

Advantage of this measure is that it can take dinucleotide biases that
exist in the genomes into account. However, the great disadvantage is
that this measure will be influenced by conservation(consensus) of the
sequences. Also it seems that it is not appropriate to consider only
first ordered Markov chain.

\subsection{Statistical independency test}

The second way to evaluate correlation between bases is to assume that
probability of observing any base b1 at position p1 has no effect on
probability of observing any base b2 at position p2.

To formulate it statistically, let \(P(b1)\) be probability of
observing base b1 at position p1, and \(P(b2)\) be probability of
observing base b2 at position p2. And let \(P(b1, b2)\) be probability
of observing base b1 at position p1, and base b2 at position p2 at the
same time. If position p1 and p2 have no correlation to each other,
then the following formula holds.

\[ P(b1, b2) = P(b1)P(b2) \quad \mbox{for all } b1,b2=a,t,c,g. \]


And this is the null hypothesis. Alternative hypothesis is

\[ P(b1, b2) \neq P(b1)P(b2) \quad \mbox{for some }b1,b2=a,t,c,g. \]

To evaluate statistical significance, the following formula is used.

\[
2N\sum_{i=a,t,c,g}\sum_{j=a,t,c,g}
                  P(i,j)\log\frac{P(i,j)}{P(i)P(j)} \quad \sim \chi^{2} \:\: \mbox{with 9 degrees of freedom}
\]

Advantage of using this measure is that it will not be influenced by
base conservation. However, since this measure does not take
dinucleotide biases exist in the genome into account, large p-values
will be observed especially between two close positions.

\subsection{Mutual information}

Third way is to apply information theory. If two bases in two
positions are independently distributed, then amount of information
about one position that we obtain when we observe a base at the other
position is 0. If two bases in two positions are dependent, then we
will obtain some amount of information about one position when we
observe a base at the other position.

Mutual information(\(MI\)) measures amount of information that we obtain about
one position when we observe bases at the other position, and it is
defined by the following formula.

\[
MI\mbox{(bit)} = \sum_{i=a,t,c,g}\sum_{j=a,t,c,g}P(i,j)\log_{2}\frac{P(i,j)}{P(i)P(j)}
\]

\noindent
where \(P(i)\) denotes probability of observing base \(i\) at position
p1, \(P(j)\) denotes probability of observing base \(j\) at position
p2. And \(P(i,j)\) denotes probability of observing base \(i\) at
position p1 and observing base \(j\) at position p2 at the same time.
It can shown that

\[ 0 \leq MI \leq 2 \]

\(MI\) will be close to 2 as position p1 and p2 become dependent. 
This measure will not be influenced by conservation of sequences. 
Moreover, this measure will not be influenced by number of sequences to 
analyze. Thus, it is possible to compare \(MI\)s of different pairs of 
positions. 


\subsection{Other methods}

There are several other ways to measure correlations in the sequences. 
For example, Herzel et al.\cite{label511} have applied autocorrelation
functions to sequences. Unlike mutual information, it shows more
information about which base has correlation with which base. In
addition, Fourier transformation may be useful to evaluate long range
correlations.

In this work, measurements such as O/E ratio of observation of each
bases are used to specify correlations.

\subsection{Minimum length of intergenic regions}

Procaryotic mRNAs are usually polycistronic. In many cases, genes next
to each other are very close to each other. In this case, translation
initiation sites are overlapped by upstream coding regions, which may
have correlation between their codons.  To avoid it, only sequences
whose upstream genes are more than 200 bases apart are used for the
analysis of procaryotic sequences.


\section{Regions to analyze}

Analyses was concentrated on the following three kinds of correlations.

\begin{description}
\item[Analysis of adjacent bases surrounding translation initiation
sites] This will give whether bases in consensus sequences are formed
independently, or there is greater natural selection to form specific
sequences such as Shine-Dalgarno sequences.
\item[Analysis of correlation between upstream region and downstream
region] Many organisms have consensus sequences in both upstream and
downstream regions(coding region). However, how they interact is unclear.
This analysis will give prospective on relationship between consensus
sequences located upstream and downstream.
\item[Analysis of correlation between start codons and other regions in
bacteria] Start codons in procaryotes are AUG, GUG or UUG. However,
effect of type of start codons on the other regions, such as
Shine-Dalgarno sequence is unclear(It was also discussed in chapter
\ref{chiyo}). Thus this analysis will give relationship between type of
start codon and other consensus sequences.
\end{description}

\section{Results and Discussion}


\subsection{Analysis of adjacent bases}
\label{adjacent_corr}

Test of Markov process was applied to translation initiation sites
in {\it E.coli}. The result is shown in figure \ref{ecmark}. Values are
indicated by \(\chi^{2}\).

\begin{figure}
\begin{center}
\epsfile{file=ecmark1.eps,scale=0.7}
\epsfile{file=ecmark4.eps,scale=0.7}
\end{center}
\caption{Test on first Markov process in {\it E.coli}}
\label{ecmark}
\end{figure}

As this measurement is influenced by conservation of the sequences,
peaks observed in \ref{ecmark} show Shine-Dalgarno sequences and start
codons. Furthermore, the values are always at significant level,
suggesting that modeling sequences by first Markov process is not
appropriate way. Thus, tests using first Markov process will not be
applied further.

Instead dependency test can be used. However, as genome have
dinucleotide bias, significant values will be observed
everywhere. Thus, mutual information will be used instead.

\begin{figure}
\begin{center}
\epsfile{file=ecolimut1.eps}
\epsfile{file=ecolimut2.eps}
\end{center}
\caption{Mutual information in translation initiation sites in {\it E.coli}}
\label{miec1}
\end{figure}


\begin{figure}
\begin{center}
\epsfile{file=ecolimut3.ps,scale=0.4}
\epsfile{file=ecolimut4.ps,scale=0.4}\\
\epsfile{file=ecolimut5.ps,scale=0.4}
\epsfile{file=ecolimut6.ps,scale=0.4}
\end{center}
\caption{Mutual information in translation initiation sites in {\it E.coli}(More than 2 bases apart)}
\label{miec3}
\end{figure}

The result for {\it E.coli} is shown in figure \ref{miec1} and
\ref{miec3}.  Obviously, there is high level correlation at positions
where Shine-Dalgarno sequences exist. Correlations seems to extend to 5
bases. Analyses of other procaryotes also showed that in most
procaryotes, there are high level correlations at positions where
Shine-Dalgarno sequences exist. In {\it E.coli}, correlation between two
positions which are 2 bases apart had the greatest correlation.  However
about half of procaryotes that were analyzed so far had greatest
correlations in Shine-Dalgarno sequences when two positions were just
one base apart. In {\it A.fulgidus}, {\it A.aeolicus} and {\it
T.maritima}, two positions which are 3 bases apart had the greatest
correlation, and in {\it M.tuberculosis}, two positions which are 4
bases apart had the greatest correlation.

To find which base has correlation with which base, we have calculated
``independence O/E'' for each pair of bases. Independence O/E of base
\(i\) and \(j\) is calculated by the following formula.

\[
\mbox{Independence O/E of base \(i\) and \(j\)} = \frac{P(i, j)}{P(i)P(j)}
\]

\noindent
where \(P(i,j)\) denotes ratio of pairs with base \(i\) at position
p1, and base \(j\) at position p2. And \(P(i)\) denotes ratio of base
\(i\) at position p1, and \(P(j)\) denotes ratio of base \(j\) at
position p2, respectively. Results of correlation with A and G which are one
base apart are shown in figure \ref{ecioea-} and \ref{ecioeg-}.


\begin{figure}
\begin{center}
\epsfile{file=ecaa_oe.ps,scale=0.4}
\epsfile{file=ecac_oe.ps,scale=0.4}
\epsfile{file=ecag_oe.ps,scale=0.4}
\epsfile{file=ecat_oe.ps,scale=0.4}
\end{center}
\caption{Independence O/E of A- in {\it E.coli}}
\label{ecioea-}
\end{figure}

\begin{figure}
\begin{center}
\epsfile{file=ecga_oe.ps,scale=0.4}
\epsfile{file=ecgc_oe.ps,scale=0.4}
\epsfile{file=ecgg_oe.ps,scale=0.4}
\epsfile{file=ecgt_oe.ps,scale=0.4}
\end{center}
\caption{Independence O/E of G- in {\it E.coli}}
\label{ecioeg-}
\end{figure}

Recall that core Shine-Dalgarno sequence is ``AGGAGG''. There is high
positive correlation of ``AG'', ``GA'' and slight positive correlation
of ``GG'' at ribosome binding sites. From these results, bases that
constitute Shine-Dalgarno sequences is not formed independently by
molecular evolution. Molecular evolutions may have occurred so as to form
sequence patterns of Shine-Dalgarno sequence. This can be explained from
the following point of view. Single bases cannot contribute to bindings
of 16S rRNA, because bindings occur according to the amount of free
energy for that bindings, and single base has little effect for free
energy. However, when bases are dinucleotide or triplet, it becomes
easier for 16S rRNA to bind to it, and contribution to translation
initiation becomes very large. Thus, when there is A at one position,
molecular evolution may work to make G at the next position.

\begin{figure}
\begin{center}
\epsfile{file=hsapmut1.eps,scale=0.7}
\end{center}
\caption{Mutual information in translation initiation sites in {\it H.sapiens}}
\label{hsapmut1}
\end{figure}

\begin{figure}
\begin{center}
\epsfile{file=mmusmut1.eps,scale=0.7}
\end{center}
\caption{Mutual information in translation initiation sites in {\it M.musclus}}
\label{mmusmut1}
\end{figure}

The results for {\it H.sapiens} and {\it M.musclus} are shown in figure
\ref{hsapmut1} and \ref{mmusmut1}.  In both cases, there is high level
of correlation only between position -2 and -1. However, the core
position of Kozak's consensus is position -3, suggesting that this
correlation does not have strong effect on forming Kozak's
consensus pattern. To find which base has correlation with which base,
``dinucleotide matrix'', and ``dinucleotide O/E matrix'' have been
constructed, which are defined by the following formula.

\vspace{1em}

\begin{math}
\mbox{Dinucleotide matrix}\:\:(a_{ij}) \quad
a_{ij} = \frac{P(i,j)}{P(i)}
\end{math}

\begin{math}
\mbox{Dinucleotide O/E matrix}\:\:(b_{ij}) \quad
b_{ij} = \frac{P(i,j)}{P(i)P(j)}
\end{math}

\vspace{1em}

\noindent
where \(P(i,j)\) denotes ratio of pairs with base \(i\) at position
p1 and base \(j\) at position p2, \(P(i)\) denotes ratio of base
\(i\) at position p1, and \(P(j)\) denotes ratio of base \(j\) at
position p2, respectively. Results for {\it H.sapiens} are shown in
figure \ref{dinuc_r_mat_hsap} and \ref{dinuc_oe_mat_hsap}, and those
for {\it M.musclus} are shown in figure \ref{dinuc_r_mat_mmus} and
\ref{dinuc_oe_mat_mmus}.


\begin{table}
\begin{center}
\begin{tabular}{l|llll}
  & a & t & c & g \\
\hline
a &0.27&0.05&0.27& 0.41\\ 
t &0.13&0.14&0.54& 0.20\\ 
c &0.13&0.07&{\bf 0.66}& 0.13\\
g &0.22&0.04&0.36& 0.37\\
\end{tabular} 
\end{center}
\caption{Dinucleotide ratio matrix between position -2 and -1 in {\it H.sapiens}}
\label{dinuc_r_mat_hsap}
\end{table}


\begin{table}
\begin{center}
\begin{tabular}{l|llll}
    & a  & t  & c  & g  \\
\hline
a & {\bf 1.42} & 0.73& 0.57& 1.55\\ 
t & 0.69 & {\bf 1.96} & 1.12& 0.74\\ 
c & 0.72 & 1.05& {\bf 1.37} & 0.50\\ 
g & 1.19 & 0.63& 0.75& {\bf 1.42}\\
\end{tabular}
\end{center}
\caption{Dinucleotide O/E matrix between position -2 and -1 in {\it H.sapiens}}
\label{dinuc_oe_mat_hsap}
\end{table}


\begin{table}
\begin{center}
\begin{tabular}{l|llll}
    & a  & t  & c  & g  \\
\hline
a   &0.26&0.06&0.26&0.43\\ 
t   &0.12&0.10&0.60&0.18\\ 
c   &0.11&0.08&{\bf 0.68}&0.13\\ 
g   &0.20&0.07&0.35&0.38\\
\end{tabular}
\end{center} 
\caption{Dinucleotide ratio matrix between position -2 and -1 in {\it M.musclus}}
\label{dinuc_r_mat_mmus}
\end{table}


\begin{table}
\begin{center}
\begin{tabular}{l|llll}
    & a  & t  & c  & g  \\
\hline
a  & {\bf 1.50}& 0.75 &0.54 &1.58\\ 
t  & 0.70& {\bf 1.38} &1.23 &0.66\\ 
c  & 0.63& 1.09 &{\bf 1.41} &0.48\\ 
g  & 1.19& 0.95 &0.71 &{\bf 1.41}\\
\end{tabular}
\end{center}
\caption{Dinucleotide O/E matrix between position -2 and -1 in {\it M.musclus}}
\label{dinuc_oe_mat_mmus}
\end{table}


Recall that nucleotide pattern around start codons are known as Kozak's
consensus, i.e. GCC$^{\rm A}_{\rm G}${\bf
CC}\underline{ATG}G\cite{label3}. Thus, in dinucleotide matrix, element
corresponding to CC is high. However, according to the figures
\ref{dinuc_oe_mat_hsap} and \ref{dinuc_oe_mat_mmus}, values
corresponding to ``AA'',''TT'', ``CC'' and ``GG'' are obviously
high. Thus, it can be inferred that molecular evolutions worked to make
bases at position -2 and -1 the same. There is no clear explanation for
this observation. Table \ref{hsap_18S_oe} shows O/E value of number of
dinucleotides in 18S rRNA of {\it H.sapiens}(O/E = rRNA length x
N(b1b2)/(N(b1) x N(b2)), where N(b1b2) is number of dinucleotides b1b2
observed in 18S rRNA, and N(b) is the number of nucleotides b observed in
18S rRNA). Relatively high O/E values are observed when two nucleotides
are the same. This may suggest that although not specific region, many
regions in 18S rRNA may be likely to bind to position -2 of mRNA.

\begin{table}
\noindent
%Length of 18S rRNA = 1868
%Base content: a 419;  t 402;  c 498;  g 549;
\begin{center}
\begin{tabular}{lr|lr|lr|lr}
aa & {\bf 1.23} &  at & 1.05 & ac & 0.85 & ag & 0.91 \\
ta & 0.92 &  tt & {\bf 1.29} & tc & 0.91 & tg & 0.93 \\
ca & 0.78 &  ct & 0.90 & cc & {\bf 1.23} & cg & 1.03 \\
ga & 1.08 &  gt & 0.83 & gc & 0.97 & gg & {\bf 1.08} \\
\end{tabular}
\end{center}
\caption{O/E value of number of dinucleotides in 18S rRNA of {\it H.sapiens}}
\label{hsap_18S_oe}
\end{table}

Analyses of other eucaryotes were also conducted. In {\it
D.melanogaster}, {\it A.thaliana} and {\it S.cerevisiae}, no peak of
correlations before start codons were found(Only result for {\it
S.cerevisiae} is shown in figure \ref{Scer_minf1}). In {\it C.elegans},
slight peak of correlations is found in the region ranging approximately
from -10 to -5(Figure \ref{Cele_minf1}).

\begin{figure}
\begin{center}
\epsfile{file=scermut1.eps,scale=0.7}
\end{center}
\caption{Mutual information in translation initiation sites in {\it S.cerevisiae}}
\label{Scer_minf1}
\end{figure}

\begin{figure}
\begin{center}
\epsfile{file=celemut1.eps,scale=0.7}
\end{center}
\caption{Mutual information in translation initiation sites in {\it C.elegans}}
\label{Cele_minf1}
\end{figure}

Why do most procaryotes have correlation in 5'UTR whereas only part of
eucaryotes have correlation in 5'UTR? As mentioned above, procaryotes
have correlation in 5'UTR presumably because 16S rRNA binds mRNA.
However, does 18S rRNA bind mRNA to initiate translation in eucaryotes? 
Sequences at 3' terminal of 18S rRNA seem to be conserved in many
eucaryotes, and they have some homology with 3' terminal sequence of
{\it E.coli}\cite{label7078}. And some statistical analysis have shown
that in some mRNA, there are sequence patterns in 5'UTR which are
complement to 18S rRNA 3' terminal
sequences\cite{label7098}. Furthermore, biological experiments have
shown that mRNA and 18S rRNA have some interaction with each
other\cite{label7551}. In addition, above work showed that 18S rRNA may
bind to position -2 of mRNA in {\it H.sapiens} and {\it
M.musclus}. Thus, it may be true that in some species, there is
interaction between some regions of 18S rRNA and some regions of
mRNA. However, direct interaction between 18S rRNA and consensus
sequence on mRNA(as in the case of Shine-Dalgarno sequence and 16S rRNA
in procaryotes) do not seem to be universal mechanism of translation
initiation in eucaryotes, because there is few evidence that this kind
of interaction is absolute requirement of translation initiation in most
eucaryotes. Thus, this may be reason why correlation is not observed in
all the eucaryotes.

\subsection{Analysis of upstream and downstream consensus}

Mutual information of all the pairs of positions ranging from -20 to
+10 were calculated. The result for {\it E.coli} is shown in figure
\ref{ecoliud}. Rectangular area indicates mutual information between
upstream region and downstream region.

\begin{figure}
\begin{center}
\epsfile{file=ecoliud.eps}
\end{center}
\caption{Mutual information between upstream and downstream in {\it E.coli}}
\label{ecoliud}
\begin{quotation}
\begin{small}
Numbers at the bottom correspond to pairs of positions where mutual
 information is calculated. Height indicates mutual information. 
Rectangular area indicates mutual information between upstream region
 and downstream region.
\end{small}
\end{quotation}
\end{figure}

The result shows that mutual information is relatively high in diagonal
area, which implies that mutual information is relatively high between
closely positioned bases. Especially, in the region where Shine-Dalgarno
sequences may exist, higher mutual information is observed, which we
have already discussed in \ref{adjacent_corr}. However, there seems to
be no obvious correlation between upstream region and downstream region.

\begin{figure}
\begin{center}
\epsfile{file=hsapud.eps}
\end{center}
\caption{Mutual information between upstream and downstream in {\it H.sapiens}}
\label{hsapud}
\end{figure}

The result for {\it H.sapiens} is shown in figure \ref{hsapud}.
As in the case of {\it E.coli}, mutual information is relatively high
between closely positioned bases, especially between position -2 and -1
as discussed in \ref{adjacent_corr}. However, there seems to be no
obvious correlation between upstream region and downstream region.

Analyses of correlations between upstream and downstream regions in
other eucaryotes were also conducted. In {\it C.elegans}, {\it
D.melanogaster}, {\it A.thaliana} and {\it S.cerevisiae}, no strong
correlations between upstream and downstream regions were found.

From these results, there seems to be no correlation between upstream
and downstream, or correlation seems to be very weak. To test whether
there is correlation between upstream and downstream region,
independency test was conducted. Test was concentrated on correlations
between position +3 and upstream region, because position +3 seems to be
most conserved position downstream which is important for translation
initiation.

\begin{figure}
\begin{center}
\epsfile{file=ecoliupd3.eps,scale=0.7}
\end{center}
\caption{Test on correlation with position +3 in {\it E.coli}}
\label{ecoliupd}
\end{figure}

The result for {\it E.coli} is shown in figure \ref{ecoliupd}.  There
is significant correlation between -1 and +3. However, this may be
because -1 and +3 are close to each other, or as those two positions
are beside start codon, they may have influence on initiator-tRNA.

\begin{figure}
\begin{center}
\epsfile{file=hsapupd3.eps,scale=0.7}
\end{center}
\caption{Test on correlation with position +3 in {\it H.sapiens}}
\label{hsapupd}
\end{figure}

The result for {\it H.sapiens} is shown in figure \ref{hsapupd}.
There is obvious correlation between position -3 and +3, which are
Kozak's core consensus.

From these results, there may be some correlations
between upstream region and downstream region, but they are very weak.

\subsection{Analysis of start codons}

In this analysis, correlation between the first base of start codon and
other bases are measured using independency test. The result for {\it E.coli}
is shown in figure \ref{ecstart}.

\begin{figure}
\begin{center}
\epsfile{file=ecstart2.eps}
\end{center}
\caption{Correlation with start codons in {\it E.coli}}
\label{ecstart}
\end{figure}

There is slightly significant correlation with base at position -9,
which can be inferred as Shine-Dalgarno region. Thus there is probability
that start codon correlates with Shine-Dalgarno sequence, and they do
not work independently as recognition sequence. There are peaks at +4
suggesting that this position may also correlate with start codon.

There is significant peak at position -44. There is no clear explanation
for this observation. The dinucleotide O/E matrix between -44 and the
first base of start codons is shown in table \ref{ecstart_44_0}.

\begin{table}
\begin{center}
\begin{tabular}{c|lll}
  &  a  & t   &  g\\
\hline
a & 1.04 & {\bf 1.61} & 0.62\\
t & 0.97 & 1.35 & 1.09\\
c & 0.97 & 0.36 & {\bf 1.35}\\ 
g & 1.02 & 0.39 & 1.03\\
\end{tabular}
\end{center} 
\caption{Dinucleotide O/E matrix between -44 and the first bases of
 start codons in {\it E.coli}}
\label{ecstart_44_0}
\end{table}

When start codons are not AUG, there are comparatively high correlations 
of A-T and C-G. Thus one possible explanation is that position -44 and
start codon may be involved in the formation of secondary structure when 
the start codon is not AUG.

\section{Summary}

Calculations of correlations that exist in translation initiation sites
were conducted from three points of view, i.e. correlation between
adjacent bases, correlation between upstream and downstream regions,
correlation between start codons and surrounding bases. In procaryotes,
correlations between adjacent bases become strong in ribosome binding
sites presumably to form Shine-Dalgarno sequences.  Correlations between
the first bases of start codons and Shine-Dalgarno sequences were also
found. In eucaryotes, there was no clear tendency of correlation to form
consensus sequences around start codons, presumably because in
eucaryotes in general, 18S rRNA does not bind to consensus sequence on
mRNA to initiate translation. However, in {\it H.sapiens} and {\it
M.musclus}, correlations between position - 2 and - 1 were
discovered. These correlations are likely to make two bases at these
positions the same. In the species that were analyzed so far,
correlations between upstream region and downstream region seem to be
weak.

