<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<link href="css/main.css" type="text/css" rel="stylesheet" media="screen">
<title>What is Cytoscape?</title>
<script type="text/javascript" 
src="http://ajax.googleapis.com/ajax/libs/jquery/1.4.2/jquery.min.js"></script>
<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jqueryui/1.8.1/jquery-ui.min.js"></script>
<script type="text/javascript" src="js/menu_generator.js"></script>
</head>

<body>
<div id="container">
<script src="js/header.js"></script>

<div id="introtitle"> <img src="images/title3.png"/> </div>

<div class="blockfull">
  <h2>What is Cytoscape?</h2>
  <img src="images/intro_2.png" /><img src="images/intro_1.png" />
  <p> <strong><i>Cytoscape</i></strong> is an open source bioinformatics software platform for <strong><i> visualizing</i></strong> molecular interaction networks and biological pathways and <strong><i>integrating</i></strong> these networks with annotations, gene expression 
    profiles and other state data. &nbsp; Although Cytoscape was originally designed for 
    biological research, now it is a general platform for complex network analysis and 
    visualization. &nbsp;  Cytoscape <i>core</i> distribution provides a basic set of features 
    for data integration and visualization. &nbsp; Additional features are available as <strong><i>plugins</i></strong>. &nbsp; Plugins are available for network and molecular 
    profiling analyses, new layouts, additional file format support, scripting, and connection 
    with databases. &nbsp; Plugins may be developed by anyone using the Cytoscape open API based on <a href="http://www.java.com/en/" target="_blank">Java&#8482;</a> technology and plugin <a href="community.php"> community development</a> is encouraged.  Most of the 
    plugins are <a href="plugins/index.php">freely available</a>. </p>
</div>

  <div class="blockleft">
    <h4>Supports Many Standards</h4>
    <img src="images/intro_3.png" />
    <p> Cytoscape supports a lot of standard network and annotation file formats including: SIF 
      (Simple Interaction Format), <a href="http://www.infosun.fim.uni-passau.de/Graphlet/GML/"> GML</a>, <a href="http://www.cs.rpi.edu/~puninj/XGMML/">XGMML</a>, <a href="http://www.biopax.org/">BioPAX</a>, <a href="http://www.psidev.info/index.php?q=node/60">PSI-MI</a>, <a href="http://sbml.org/Main_Page">SBML</a>, <a href="http://www.obofoundry.org/">OBO</a>, 
      and <a href="http://www.geneontology.org/GO.format.annotation.shtml">Gene Association</a>.  
      Delimited text files and MS Excel&#8482; Workbook are also supported and you can import 
      data files, such as expression profiles or <a href="http://www.geneontology.org">GO</a> annotations, generated by other applications 
      or spreadsheet programs.  Using this feature, you can load and save arbitrary attributes 
      on nodes, edges, and networks.  For example, input a set of custom annotation terms for 
      your proteins, create a set of confidence values for your protein-protein interactions. </p>
  </div>
  <div class="blockright">
    <h4>Web Service Clients</h4>
    <img src="images/intro_4.png" />
    <p> Cytoscape works as a web service client.  This means Cytoscape can directly connect to 
      the external public databases and imports network and annotation data.  Currently, <a href="http://www.pathwaycommons.org/pc/">Pathway Commons</a>, <a href="http://www.ebi.ac.uk/intact/site/index.jsf">IntAct</a>, <a href="http://www.biomart.org/">BioMart</a>, <a href="http://www.ncbi.nlm.nih.gov/sites/entrez?db=gene">NCBI Entrez Gene</a>, 
      and <a href="http://www.ebi.ac.uk/Tools/picr/">PICR</a> are supported.  And we continue to 
      develop new service clients for popular databases. </p>
  </div>
  <div class="blockright">
    <h4>Interoperability</h4>
    <img src="images/intro_5.png" />
    <p> Since Cytoscape supports import/export standard file formats, you can easily put Cytoscape 
      into your workflow.  For example, if you have a network data generated by <a href="http://cneurocvs.rmki.kfki.hu/igraph/">igraph</a> or <a href="http://www.bioconductor.org/packages/release/GraphsAndNetworks.html">Bioconductor</a>, 
      Cytoscape can load the file as a text table and you can export it in PSI-MI format for 
      other bioinformatics tools or your own programs/scripts. </p>
  </div>
  <div class="blockleft">
    <h4>Session File</h4>
    <img src="images/intro_6.png" />
    <p> You can save your work by a single click.  All of settings, data files, and visualizations 
      are packed in a session file.  It is called <i>Cytoscape Session (.cys) file</i>.  
      Cytoscape Session file includes networks, attributes (for node/edge/network), Desktop 
      states (selected/hidden nodes and edges, window sizes), Properties, some plugin states, 
      and Visual Styles. </p>
  </div>
  <div class="blockleft">
    <h4>VizMapper&#8482;</h4>
    <img src="images/intro_12.png" /> <img src="images/intro_7.png" />
    <p> Customize network data display using powerful <a href="http://cytoscape.wodaklab.org/wiki/Cytoscape_User_Manual/Visual_Styles"> <strong><i>Visual Styles</i></strong></a>.  
      View a superposition of gene expression ratios and p-values on the network. &nbsp;
      Expression data can be mapped to node color, label, border thickness, or border color, 
      etc. according to user-configurable colors and visualization schemes. </p>
  </div>
  <div class="blockright">
    <h4>Image Export</h4>
    <img src="images/pdficon_large.gif" />
    <p> You can export networks as publishiable-quality images.  Supports PDF, EPS, SVG, PNG, 
      JPEG, and BMP files.  Vector images (PDF and EPS) can be modified by other application 
      such as Adobe Illustrator for further enhancements. </p>
  </div>
  <div class="blockright">
    <h4>Layout</h4>
    <img src="images/intro_9.png" />
    <p> Layout networks in two dimensions. &nbsp;A variety of layout algorithms are available, 
      including cyclic, tree, force-directed, edge-weight, and yFiles Organic layouts.  You can 
      also use Manual Layout tools similar to other graphics application user interface. </p>
  </div>
  <div class="blockright">
    <h4>Browsing</h4>
    <img src="images/intro_11.png" />
    <p> Zoom in/out and pan for browsing the network.  Use the network manager to easily organize 
      multiple networks.  And this structure can be saved in a session file.  Use the <i>bird's 
      eye view</i> to easily navigate large networks.  Easily navigate large networks (100,000+ nodes and edges) by efficient rendering engine. </p>
  </div>
  
  <div class="blockleft">
    <h4>Search</h4>
    <img src="images/intro_8.png" /> <img src="images/intro_17.png" />
    <p> <strong>Search target nodes and edges</strong> by Quick Find or Enhanced Search. <a href="http://conklinwolf.ucsf.edu/genmappwiki/Google_Summer_of_Code_2007/Maital"> Lucene Syntax</a> is supported by <i>Enhanced Search Plugin (ESP)</i> for arbitrary complex queries. </p>
  </div>
  
  <div class="blockleft">
    <h4>Filter</h4>
    <img src="images/intro_16.png" />
    <p> <strong>Filter the network</strong> to select subsets of nodes and/or interactions based 
      on the current data. &nbsp;For instance, users may select nodes involved in a threshold 
      number of interactions, nodes that share a particular GO annotation, or nodes whose gene 
      expression levels change significantly in one or more conditions according to p-values 
      loaded with the gene expression data.  You can create new networks from the filtering result. </p>
  </div>
  <div class="blockright">
    <h4>Find Modules/ Clusters</h4>
    <img src="images/intro_14.png" /> <img src="images/intro_15.png" />
    <p> <strong>Find active subnetworks/pathway modules</strong>. The network is screened against gene expression data to identify connected sets of interactions, i.e. <i>interaction subnetworks</i>, whose genes show particularly high levels of differential expression. &nbsp;The interactions contained in each subnetwork provide hypotheses for the regulatory and signaling interactions in control of the observed expression changes. <strong>Find clusters (highly interconnected regions)</strong> in any network loaded into Cytoscape. Depending on the type of network, clusters may mean different things. For instance, clusters in a protein-protein interaction network have been shown to be protein complexes and parts of pathways. Clusters in a protein similarity network represent protein families. </p>
  </div>
  <div class="blockleft">
    <h4>Plugin Manager</h4>
    <img src="images/intro_18.png" />
    <p> <strong><a href="plugins2.php">Plugins</a></strong> available for network and molecular 
      profile analysis.  Cytoscape is a software written in Java and you can write your own 
      plugin for data analysis/import/visualization by writing Java code.  More plugins 
      available on the <a href="plugins2.php">plugins page</a>.  You can isntall most of the  
      plugins by just one click from <i>Plugin Manager</i>. </p>
  </div>
  <div class="blockright">
    <h4>Multilingual</h4>
    <img src="images/intro_13.png" />
    <p> You can use your language for the data files.  Most of the features in Cytoscape supports 
      languages other than English, including Eastan Asian languages using two-byte characters. </p>
  </div>
  
  <div class="blockabout">
    <h2>History and Future</h2>
    <p> Cytoscape was initially made public in July, 2002 (v0.8); the second release (v0.9) was in 
      November, 2002. and v1.0 was released in March 2003.&nbsp; Version 1.1.1 is the last stable 
      release for the 1.0 series.&nbsp; Latest version of Cytoscape is
      <?= $latest_version ?>
      : <a href='<?= $latest_download_link ?>'>download</a> (<a href='<?= $latest_release_notes_link ?>'> release notes</a> ).&nbsp;
      
      Cytoscape Core developer team continue to work on this project and near future, we are going to 
      release next major version, Cytoscape 3.0.  It will be more modularized and scalable version 
      of Cytoscape.  This release includes server-side features.  For more information about version 
      3.0, please visit <a href="http://cytoscape.wodaklab.org/wiki/Cytoscape_3.0">developer's 
      wiki page</a>. </p>
  </div>
  <script src="js/footer.js"></script>
  </div>
  
</body>
</html>
